internal/jsonpathwriter/jsonpathwriter.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jsonpathwriter
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/PaesslerAG/jsonpath"
+)
+
+func Print(w io.Writer, path string, obj interface{}) error {
+	jsonString, err := json.Marshal(obj)
+	if err != nil {
+		return err
+	}
+
+	val := interface{}(nil)
+	err = json.Unmarshal(jsonString, &val)
```

Comment:
[nit] this is the right approach given json path limitations but you can optimize this a bit
```suggestion
	var val interface{}
	err = json.Unmarshal(jsonString, &val)
```
Saves you an instaciation you won't need later, also you could group this in the if `if err2 := json.Unmarshal(...` kind of thing but that's optional